pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -94,7 +94,7 @@ def __init__(
         amp_type: str = "native",
         amp_level: Optional[str] = None,
         sync_batchnorm: bool = False,
-        benchmark: bool = False,
+        benchmark: Optional[bool] = None,
         replace_sampler_ddp: bool = True,
         deterministic: bool = False,
```

Comment:
We currently don't do anything with the Trainer(deterministic) flag. The AcceleratorConnector rewrite #11448 missed this. I had a review comment but it was missed. 

We need to actually set the deterministic flag on the torch library / cudnn library. @krshrimali could you take care of it?